config/confighttp/compression.go

```diff
@@ -58,17 +58,10 @@ var availableDecoders = map[string]func(body io.ReadCloser) (io.ReadCloser, erro
 		}
 		return zr, nil
 	},
+	//nolint:unparam // ignoring lint since method needs to fit typedef
 	"snappy": func(body io.ReadCloser) (io.ReadCloser, error) {
-		sr := snappy.NewReader(body)
-		sb := new(bytes.Buffer)
-		_, err := io.Copy(sb, sr)
-		if err != nil {
-			return nil, err
-		}
-		if err = body.Close(); err != nil {
-			return nil, err
-		}
-		return io.NopCloser(sb), nil
+		// Lazy Reading content to improve memory efficiency
+		return io.NopCloser(snappy.NewReader(body)), nil
```

Comment:
No longer possible to use `NopCloser`. Need to close the body when closing the reader correct?

Reply:
Yes, technically when the new body is created, the old body should be drained and closed since compression readers make no promises that they perform the same actions on the underlying buffers.

Reply:
Still, you can return a `ReadCloser` that closes the Body when closed correct?

Reply:
Sounds reasonable, it would need to be a custom wrapper since I don't believe there is anything in the standard library that currently does this?

Do you think it makes sense to implement this in a seperate PR so it can be applied to the existing readers?

Reply:
> Do you think it makes sense to implement this in a seperate PR so it can be applied to the existing readers?
Yes, but that PR has to go first, since this PR downgrades the behavior otherwise, correct?

You can have a simple implementation
```golang
type readCloser struct {
	io.Reader
	body http.Body
}

func (rc readCloser) Close() error {
  return rc.body.Close()
}

// Lazy Reading content to improve memory efficiency
return readCloser{Reader: snappy.NewReader(body), body: body}, nil
```